pkg/rke2/registries.go

```diff
@@ -45,85 +47,96 @@ func GenerateRegistries(rke2ConfigRegistry RegistryScope) (*Registry, []bootstra
 	}
 
 	for configName, regConfig := range rke2ConfigRegistry.Registry.Configs {
-		tlsSecret := corev1.Secret{}
-		authSecret := corev1.Secret{}
-
-		err := rke2ConfigRegistry.Client.Get(
-			rke2ConfigRegistry.Ctx,
-			types.NamespacedName{
-				Name:      regConfig.TLS.TLSConfigSecret.Name,
-				Namespace: regConfig.TLS.TLSConfigSecret.Namespace,
-			},
-			&tlsSecret,
-		)
-		if err != nil {
-			rke2ConfigRegistry.Logger.Error(err, "TLS Config Secret for the registry was not found!")
-
-			return &Registry{}, []bootstrapv1.File{}, err
+		registryConfig := RegistryConfig{}
+
+		if regConfig.TLS == (bootstrapv1.TLSConfig{}) && regConfig.TLS != (bootstrapv1.TLSConfig{}) {
```

Comment:
Its best to put this type of check into the "validating webhook" so that its validated when the user tries to apply the manifest.

The check would need to be in a coupld of places:

- [ValidateCreate](https://github.com/rancher-sandbox/cluster-api-provider-rke2/blob/e828ffa7bd9c300b7d7ced430b64e6bcb7c94d9f/bootstrap/api/v1beta1/rke2config_webhook.go#L67)
- [ValidateUpdate](https://github.com/rancher-sandbox/cluster-api-provider-rke2/blob/e828ffa7bd9c300b7d7ced430b64e6bcb7c94d9f/bootstrap/api/v1beta1/rke2config_webhook.go#L82)